adapters/webhdfs-adapter/src/test/java/com/connexta/replication/adapters/webhdfs/WebHdfsClientTest.java

```diff
@@ -109,6 +115,94 @@ public void testStoppedClusterIsNotAvailable() throws Exception {
+    Metadata resourceMetadata = createStorageRequest.getResources().get(0).getMetadata();
+    ResourceRequest readRequest = new ResourceRequestImpl(resourceMetadata);
+    ResourceResponse readResponse = adapter.readResource(readRequest);
+    Resource readResource = readResponse.getResource();
+
+    assertThat(readResource.getId(), is(testId));
+    assertThat(readResource.getName(), is(String.format("%s_%s", testId, testDate.getTime())));
+    // TODO - uncomment below line when file content issue is fixed
+    // assertThat(readInputStreamToString(readResource.getInputStream()), is("my-data"));
+  }
+
+  @Test
+  public void testInvalidHostnameResourceUriNotRead() throws URISyntaxException {
+    String testId = "404";
+    String testName = "404-not-found";
+    Date testDate = new Date();
+    String filename = String.format("%s_%s.txt", testId, testDate.getTime());
+    String badHostnameUri = String.format("http://foobar:%s/webhdfs/v1/%s", HDFS_PORT, filename);
+
+    // creates the resource with the bad hostname
+    CreateStorageRequest createStorageRequest =
+        generateTestStorageRequest(testId, testName, testDate, testDate, new URI(badHostnameUri));
+    // Note: it's not necessary to actually write the file in HDFS
+
+    // gets the resource metadata to create a read request
+    Metadata resourceMetadata = createStorageRequest.getResources().get(0).getMetadata();
+    ResourceRequest readRequest = new ResourceRequestImpl(resourceMetadata);
+    ResourceResponse readResponse = adapter.readResource(readRequest);
+
+    assertThat(readResponse, is(nullValue()));
+  }
+
+  @Test
+  public void testInvalidPortResourceUriNotRead() throws URISyntaxException {
+    String testId = "404";
+    String testName = "404-not-found";
+    Date testDate = new Date();
+    String filename = String.format("%s_%s.txt", testId, testDate.getTime());
+    String badPortUri = "http://localhost:9999/webhdfs/v1/" + filename;
+
+    // creates the resource with the bad port
+    CreateStorageRequest createStorageRequest =
+        generateTestStorageRequest(testId, testName, testDate, testDate, new URI(badPortUri));
+    // Note: it's not necessary to actually write the file in HDFS
+
+    // gets the resource metadata to create a read request
+    Metadata resourceMetadata = createStorageRequest.getResources().get(0).getMetadata();
+    ResourceRequest readRequest = new ResourceRequestImpl(resourceMetadata);
+    ResourceResponse readResponse = adapter.readResource(readRequest);
+
+    assertThat(readResponse, is(nullValue()));
+  }
+
+  @Test
+  public void testNonExistentFileNotRead() throws URISyntaxException {
+    String testId = "404";
+    String testName = "404-not-found";
+    Date testDate = new Date();
+    String filename = String.format("%s_%s.txt", testId, testDate.getTime());
+    String filePath = "webhdfs/v1/" + filename;
+    String url = String.format("%s/%s", BASE_URL, filePath);
```

Comment:
✏️ We can probably construct the full Url in one line, since `filePath` is only being used once here